src/workflow/workflow_tasks/mod.rs

```diff
@@ -195,14 +187,16 @@ impl WorkflowTaskManager {
     ///
     /// Returns, if found, the number of attempts on the current workflow task
     pub fn request_eviction(&self, run_id: &str) -> Option<u32> {
-        if let Some(tq) = self.workflow_machines.task_queue_for(run_id) {
-            debug!(%run_id, "Eviction requested for");
-            // Queue up an eviction activation
-            self.pending_activations
-                .push(create_evict_activation(run_id.to_string()), tq);
-            let _ = self
-                .workflow_activations_update
-                .send(WfActivationUpdate::NewPendingActivation);
+        if self.workflow_machines.exists(run_id) {
+            if !self.activation_has_eviction(run_id) {
```

Comment:
I haven't read this through but can there be a case where the machine exists and has already flushed out an eviction activation?

Reply:
Not sure I follow what you mean here? Like, could we queue another one accidentally? Shouldn't be possible b/c activations can only be completed one at a time and the last one, if it was an eviction, would've destroyed the machines.